Failed build of scene_text_detection.cpp due to lack of regex support in GCC 4.8.5 (CentOS 7)

[Nightly build](http://pullrequest.opencv.org/buildbot/builders/master_etc-centos-lin64/builds/45).

```
CMakeFiles/example_dnn_scene_text_detection.dir/scene_text_detection.cpp.o: In function `main':
scene_text_detection.cpp:(.text.startup.main+0xe1a): undefined reference to `std::regex_token_iterator<__gnu_cxx::__normal_iterator<char const*, std::string>, char, std::regex_traits<char> >::regex_token_iterator()'
...
```

`regex` can be used with GCC 4.9.0+ only: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=53631

--- a/samples/dnn/scene_text_detection.cpp
+++ b/samples/dnn/scene_text_detection.cpp
@@ -1,6 +1,5 @@
 #include <iostream>
 #include <fstream>
-#include <regex>
 
 #include <opencv2/imgproc.hpp>
 #include <opencv2/highgui.hpp>
@@ -23,6 +22,21 @@ std::string keys =
         "{ evaluate e                       |false| false: predict with input images; true: evaluate on benchmarks. }"
         "{ evalDataPath edp                  | | Path to benchmarks for evaluation. "
             "Download links are provided in doc/tutorials/dnn/dnn_text_spotting/dnn_text_spotting.markdown}";
+
+static
+void split(const std::string& s, char delimiter, std::vector<std::string>& elems)
+{
+    elems.clear();
+    size_t prev_pos = 0;
+    size_t pos = 0;
+    while ((pos = s.find(delimiter, prev_pos)) != std::string::npos)
+    {
+        elems.emplace_back(s.substr(prev_pos, pos - prev_pos));
+        prev_pos = pos + 1;
+    }
+    if (prev_pos < s.size())
+        elems.emplace_back(s.substr(prev_pos, s.size() - prev_pos));
+}
 
 int main(int argc, char** argv)
 {
@@ -114,9 +128,9 @@ int main(int argc, char** argv)
                 }
                 gtLine = gtLine.substr(0, splitLoc);
 
-                std::regex delimiter(",");
-                std::vector<String> v(std::sregex_token_iterator(gtLine.begin(), gtLine.end(), delimiter, -1),
-                                      std::sregex_token_iterator());
+                std::vector<std::string> v;
+                split(gtLine, ',', v);
+
                 std::vector<int> loc;
                 std::vector<Point> pts;
                 for (auto && s : v) {

